### PR TITLE
Updated scheduler docs (tracing)

### PR DESCRIPTION
**Pull request by mskacelik**
/cc @mkouba

### CODE REVIEW

File: docs/src/main/asciidoc/scheduler-reference.adoc [hunk 1]
```diff
@@ -417,8 +417,7 @@ If the xref:smallrye-metrics.adoc[SmallRye Metrics extension] is present, then a
 
 == OpenTelemetry Tracing
 
-If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present then the `@io.opentelemetry.instrumentation.annotations.WithSpan` annotation is added automatically to every `@Scheduled` method. As a result, each execution of this method has a new `io.opentelemetry.api.trace.Span` associated.
-
+If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present, every job execution, whether using the `@Scheduled` annotation or programmatic scheduling, automatically creates a span named after the job's identity.
```
Comment (mkouba) [inline docs/src/main/asciidoc/scheduler-reference.adoc:420]:
```suggestion
If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present then every job execution, either defined with the `@Scheduled` annotation or scheduled programmatically, automatically creates a span named after the job's <<identity>>.
```

`[[identity]]` needs to be added before `=== Identity`...

Comment (mkouba) [reply docs/src/main/asciidoc/scheduler-reference.adoc]: I think that it's better to add an explicit section id if it's used in a link, e.g. something like `[#identity]`...

File: docs/src/main/asciidoc/scheduler-reference.adoc [hunk 2]
```diff
@@ -417,9 +417,8 @@ If the xref:smallrye-metrics.adoc[SmallRye Metrics extension] is present, then a
 
 == OpenTelemetry Tracing
 
-If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present then the `@io.opentelemetry.instrumentation.annotations.WithSpan` annotation is added automatically to every `@Scheduled` method. As a result, each execution of this method has a new `io.opentelemetry.api.trace.Span` associated.
+If `quarkus.scheduler.tracing.enabled` is set to `true` and the xref:opentelemetry.adoc[OpenTelemetry extension] is present then every job execution, either defined with the `@Scheduled` annotation or scheduled programmatically, automatically creates a span named after the job's <<identity>>.[[virtual_threads]]
```
Comment (mkouba) [inline docs/src/main/asciidoc/scheduler-reference.adoc:420]: The `[[virtual_threads]]` section id should be on a new line...

**Review by mkouba — APPROVED** [review]

**Comment by gsmet** [comment]
Thanks @mskacelik !

**Pull request merged**